src/idl_gen_text.cpp

```diff
@@ -161,8 +161,11 @@ template<typename T> static bool GenField(const FieldDef &fd,
                                           std::string *_text) {
   return Print(fixed ?
     reinterpret_cast<const Struct *>(table)->GetField<T>(fd.value.offset) :
-    table->GetField<T>(fd.value.offset, 0), fd.value.type, indent, nullptr,
-                                            opts, _text);
+    table->GetField<T>(fd.value.offset,
+      ("0" == fd.value.constant)
```

Comment:
What does this comparison do? If the value is `"0"` then `StringToInt` will do the right thing, so it is unnecessary.

What is necessary however is to distinguish between int and float types, since the contant may be a value like "3.14" which currently would get truncated. You'll need to inspect fd.value.type, and if its float/double, parse a float instead (`strtod`).

Reply:
actually use `NumToString` instead of `strtod`.

Reply:
Comparison with zero is fixed.
But I don't understand a problem with int and float types.
I can't create an enum with float as underlying type.
`
enum Math : float
{
  Zero=0.0,
  PI=3.14
}
table Test
{
  number : Math;
}
root_type Test;
file_identifier "TEST";
`
flatc --cpp test.fbs
test.fbs(2): error: underlying enum type must be integral